packages/mdc-segmented-button/segmented-button/foundation.ts

```diff
@@ -40,27 +40,39 @@ export class MDCSegmentedButtonFoundation extends MDCFoundation<MDCSegmentedButt
     super({...MDCSegmentedButtonFoundation.defaultAdapter, ...adapter});
   }
 
-  selectSegment(_indexOrSegmentId: number | string) {
-    return;
+  selectSegment(indexOrSegmentId: number | string) {
+    this.adapter.selectSegment(indexOrSegmentId);
   }
 
-  unselectSegment(_indexOrSegmentId: number | string) {
-    return;
+  unselectSegment(indexOrSegmentId: number | string) {
+    this.adapter.unselectSegment(indexOrSegmentId);
   }
 
   getSelectedSegments(): readonly SegmentDetail[] {
-    return [];
+    return this.adapter.getSegments().filter(segmentDetail => segmentDetail.selected);
   }
 
-  isSegmentSelected(_indexOrSegmentId: number | string): boolean {
-    return false;
+  isSegmentSelected(indexOrSegmentId: number | string): boolean {
+    let segment = this.adapter.getSegments().filter(segmentDetail => segmentDetail.index === indexOrSegmentId || segmentDetail.segmentId === indexOrSegmentId);
```

Comment:
You can also use `.some()` API (this'll exit as soon as segment index or id is found).

```ts
isSegmentSelected(indexOrSegmentId: number | string): boolean {
  return this.adapter.getSegments().some(
     segmentDetail => segmentDetail.index === indexOrSegmentId || segmentDetail.segmentId === indexOrSegmentId);
}
```